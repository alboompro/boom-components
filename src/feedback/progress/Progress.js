import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { noop } from "../../helpers";

import {
  ProgressWrapper,
  ProgressBar,
  Bar,
  ContainerBar,
  ProgressText
} from "./styles.js";

const Progress = ({
  min,
  total,
  current,
  showLabel,
  unitMeasurement,
  customLabel,
  styleProgress,
  color,
  icons,
  ...props
}) => {
  const [status, setStatus] = React.useState(0);

  const progressProps = {
    min,
    max: total,
    value: current,
    color,
    loading: props.loading,
    styleProgress
  };

  const textProps = {
    unitMeasurement,
    styleText: props.styleText
  };

  useEffect(() => {
    if (current >= total) {
      // If the progress is 100%
      setStatus(2);
    } else if (current > min && current < total) {
      // If the progress is not 100%
      setStatus(1);
    } else if (current == min) {
      // If the progress is 0%
      setStatus(0);
    }
  }, [current, total, min]);

  const getLabel = () =>
    customLabel ? (
      <span>{customLabel[status]}</span>
    ) : (
      <span>{`${current}${unitMeasurement}`}</span>
    );

  const renderLabel = () =>
    showLabel && <ProgressText {...textProps}>{getLabel()}</ProgressText>;

  return (
    <ProgressWrapper>
      <ContainerBar className="outer">
        <Bar className="inner">
          <ProgressBar {...progressProps} />
        </Bar>
      </ContainerBar>
      {renderLabel()}
    </ProgressWrapper>
  );
};

Progress.propTypes = {
  /** Minimum value of the progress bar */
  min: PropTypes.number,
  /** Total value for the progress bar */
  total: PropTypes.number,
  /** Current value of the progress bar */
  current: PropTypes.number,
  /** Current progress bar color */
  color: PropTypes.string,
  /** Enables progress bar label */
  showLabel: PropTypes.bool,
  /** Measuring unit used on progress bar label */
  unitMeasurement: PropTypes.string,
  /** Custom Nodes Array for Progress bar Label */
  customLabel: PropTypes.array,
  /** Enables progress bar with charging animation */
  loading: PropTypes.bool,
  /** Style object for progress bar */
  styleProgress: PropTypes.object,
  /** Style object for progress bar label */
  styleText: PropTypes.object
};

Progress.defaultProps = {
  min: 0,
  total: 100,
  current: 0,
  color: "royalblue",
  showLabel: false,
  unitMeasurement: "%",
  styleProgress: {},
  styleText: {}
};

export default Progress;
