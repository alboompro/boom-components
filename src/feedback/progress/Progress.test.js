import React from "react";
import { mount, shallow } from "enzyme";

import Progress from "./Progress.js";

describe("Progress", () => {
  test("Renders the component correctly", () => {
    const wrapper = shallow(<Progress current={20} color="#87CEEB" />);

    expect(wrapper).toMatchSnapshot();
  });
  test("Values of props are correct", () => {
    const wrapper = mount(
      <Progress current={20} color="#87CEEB" />
    );

    expect(wrapper.prop("current")).toBe(20);
    expect(wrapper.prop("hideLabel")).toBe(false);
    expect(wrapper.prop("color")).toEqual("#87CEEB");
  });
});
