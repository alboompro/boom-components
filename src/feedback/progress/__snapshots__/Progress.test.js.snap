// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Renders the component correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div
        color="#87CEEB"
        loading={true}
        value={20}
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <span>
      20%
    </span>
  </styled.div>
</styled.div>
`;
